### ISSUE TITLE
Show-paren matching color

### ISSUE THREAD

**Issue by ianpan870102** [purcell/color-theme-sanityinc-tomorrow#127]
Thanks for making this theme! I'm using it heavily in my Emacs workflow. However I didn't particularly like the show-paren-match background colours and plan to tweak it. I've changed the corresponding code to such:

```
      ;; Parenthesis matching (built-in)
      (show-paren-match (:foreground ,purple :background nil))
      (show-paren-mismatch (:foreground ,red :background nil))
```

But after reloading Emacs, the change still never took place. What else should I do?

**Comment by purcell** [comment]
My guess is that you changed the `.el` file but didn't byte-compile it, so Emacs is still loading the `.elc` file.

**Comment by ianpan870102** [comment]
Thanks for the reply. I'm rather an Emacs beginner, so perhaps this is why I'm confused by the compilation result:

When I ran `M-x byte-compile-file` to the color-theme-sanityinc-tomorrow.el file, I received:
```
In end of data:
color-theme-sanityinc-tomorrow.el:1429:1:Warning: the function
    ‘color-theme-install’ is not known to be defined.
```

My second question is: when I downloaded this theme, there was no `.elc` file in the folder either. Why should I byte compile it after manual tweaking?

Thanks in advance and have a nice day!

**Comment by purcell** [comment]
Ah, you didn't provide any details of how you had installed it. If there was no `.elc` previously, then byte-compilation would not be necessary.

I guess my question, then, would be: what do you mean by "reloading Emacs"? Did you completely restart it, or just re-evaluate some code?

**Comment by ianpan870102** [comment]
Thanks for getting back to me so promptly! Here's all the details I should have provided:

I'm using Emacs version 27.0.50. And I've installed your theme package through **Melpa** (`M-x package-list-packages`). Thus, the package is automatically placed in this directory: `~/.emacs.d/elpa/color-theme-sanityinc-tomorrow-20190315.239/`

**Correction**, **there are indeed** `.elc` files. I was wrong. In fact, these are all the files I have under the folder: 
```
color-theme-sanityinc-tomorrow-autoloads.el   sanityinc-tomorrow-bright-theme.elc
color-theme-sanityinc-tomorrow-pkg.el         sanityinc-tomorrow-day-theme.el
color-theme-sanityinc-tomorrow.el           sanityinc-tomorrow-day-theme.elc
color-theme-sanityinc-tomorrow.elc          sanityinc-tomorrow-eighties-theme.el
sanityinc-tomorrow-blue-theme.el            sanityinc-tomorrow-eighties-theme.elc
sanityinc-tomorrow-blue-theme.elc           sanityinc-tomorrow-night-theme.el
sanityinc-tomorrow-bright-theme.el          sanityinc-tomorrow-night-theme.elc
```

I guess my problem is then byte-compiling the `color-theme-sanityinc-tomorrow.el` file. I couldn't seem to get pass the "error": `‘color-theme-install’ is not known to be defined.`

Thanks for your time and wish you a good day!

**Comment by purcell** [comment]
Hmm. The "error" can be ignored, so let's skip that.

So the interesting thing here is that what is in `color-theme-sanityinc-tomorrow.el` (ie. the code you edited) is a big macro, which doesn't do anything on its own. In turn, each of the "sanityinc-tomorrow-*-theme.el" files contains a little piece of code which expands the macro at compilation time. So if you edited the first file (and even re-compiled it) but didn't recompile the `*-theme.el` files, then nothing will have changed in the `*-theme.elc` files, and you'll see no difference. The fix, then, is probably to delete all the `.elc` files and recompile all the `.el` files. You can easily do this via dired.

**Comment by ianpan870102** [comment]
**[ Fixed ]**
Thanks for the patient answer. I actually solved it by **deleting** all the `.elc` files altogether. (I did try to follow your way of deleting `.elc` files then **recompiling** them all via dired, but somehow it still didn't work). Now that I simply **got rid of all** the `.elc` files, I guess Emacs has no choice but to look at the real `.el` files that I tweaked (it didn't hurt my start-up time either). Anyways, thanks for the help still!
